examples/vision/mnist_moe.py

```diff
@@ -0,0 +1,480 @@
+"""
+Title: MoE for MNIST
+Author: [Damoon Shahhosseini](https://www.linkedin.com/in/damoonsh/)
+Date created: 2015/06/19
+Last modified: 2020/04/21
+Description: Simple MoE implementation for MNIST classification.
+Accelerator: GPU
+"""
+
+"""
+# Introduction
+
+In this example, we implement an adaptation of the Mixture of Experts (MoE) architecture
+([Shazeer et al.](https://arxiv.org/abs/1701.06538)).
+The idea is to use conditional computation to increases model capacity without increasing computation.
+Experts are identical blocks within a layer where each are trained to specialize in different parts of the input space.
+At each forward pass, a gating network selects a subset of experts to apply to the input.
+
+The components to implement are:
+
+- Gating network: A dense layer that outputs a probability distribution over the experts.
+- MoE layer: A layer that applies a different expert to each input in the batch. And a loss function that ensures specialization among the experts.
+- Model: A simple model that uses the MoE layer.
+"""
+
+"""
+## Imports
+"""
+
+import numpy as np
+import keras
+from keras import layers, models
+import tensorflow as tf
```

Comment:
From 2023 onwards all examples and guides added to Keras IO is Keras 3 only. Can you please update the implementation to Keras 3 - here is a guide for migration https://keras.io/guides/migrating_to_keras_3/